relay-server/src/utils/rate_limits.rs

```diff
@@ -373,6 +432,9 @@ impl Enforcement {
             profile_chunks,
         } = self;
 
+        let nested_spans = event.clone_for(DataCategory::Span, spans_length);
```

Comment:
In case `spans_length` is `0`, we will have both categories filtered out downstream since they won't be `active`.